pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/errorprone/xml/TestMethodsMustBeInTestClasses

```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
This file should have a `xml` extension

Reply:
And we need a JUnit Test class to start the tests - right now, the new rule is not tested at all....
Example: https://github.com/pmd/pmd/blob/master/pmd-apex/src/test/java/net/sourceforge/pmd/lang/apex/rule/bestpractices/ApexUnitTestClassShouldHaveAssertsTest.java
The test class is pretty empty, but still required

Docu: https://pmd.github.io/latest/pmd_userdocs_extending_testing.html

Reply:
For the sake of completeness: API version 27.0 (Spring '13) removed support for test methods inside of functional classes. I didn't realize that was so long ago, so probably the relevance even in legacy code is getting smaller and smaller...